Integrations/Elasticsearch_v2/Elasticsearch_v2.py

```diff
@@ -0,0 +1,315 @@
+    search_human_readable = tableToMarkdown('Search Metadata:', search_context, meta_headers, removeNull=True)
+    hits_human_readable = tableToMarkdown('Hits:', hit_tables, hit_headers, removeNull=True)
+    total_human_readable = search_human_readable + '\n' + hits_human_readable
+    full_context = {
+        'Elasticsearch.Search(val.Query == obj.Query && val.Index == obj.Index '
+        '&& val.Server == obj.Server && val.Page == obj.Page && val.Size == obj.Size)': search_context
+    }
+
+    return_outputs(total_human_readable, full_context, response)
+
+
+def fetch_params_check():
+    str_error = []  # type:List
+    if TIME_FIELD == '' or TIME_FIELD is None:
+        str_error.append("Index time field is not configured.")
+
+    if FETCH_INDEX == '' or FETCH_INDEX is None:
+        str_error.append("Index is not configured.")
+
+    if FETCH_QUERY == '' or FETCH_QUERY is None:
+        str_error.append("Query by which to fetch incidents is not configured.")
+
+    if TIME_FORMAT == '' or TIME_FORMAT is None:
+        str_error.append("Time format is not configured.")
+
+    if len(str_error) > 0:
+        return_error("Got the following errors in test:\nFetches incidents is enabled.\n" + '\n'.join(str_error))
+
+
+def test_general_query(es, que, test_num):
+    query = QueryString(query=que)
+    search = Search(using=es, index=FETCH_INDEX).query(query)[0:1]
+    response = search.execute().to_dict()
+    _, total_results = get_total_results(response)
+
+    if total_results == 0:
+        # failed in general query - '*'
+        if test_num == 1:
+            return_error("Fetch incidents test failed. Index value incorrect.")
+
+        # failed in getting the TIME_FIELD
+        if test_num == 2:
+            return_error("Fetch incidents test failed. Date field value incorrect.")
+
+        # failed to get the TIME_FIELD with the FETCH_QUERY
+        # this can happen and not be an error if the FETCH_QUERY doesn't have results yet.
+        # Thus this does not return an error message
+        if test_num == 3:
+            return None
+
+    else:
+        return response
+
+
+def test_func():
+    try:
+        if USERNAME:
+            res = requests.get(SERVER, auth=(USERNAME, PASSWORD), verify=INSECURE)
+        else:
+            res = requests.get(SERVER, verify=INSECURE)
+        if res.status_code >= 400:
+            return_error("Failed to connect. The following error occurred: " + HTTP_ERRORS[int(res.status_code)])
+
+    except requests.exceptions.RequestException as e:
+        return_error("Failed to connect. Check Server URL field and port number.\nError message: " + str(e))
+
+    if demisto.params().get('isFetch'):
+        # check the existence of all necessary fields for fetch
+        fetch_params_check()
+
+        try:
+            # build general Elasticsearch class
+            es = elasticsearch_builder()
+
+            # test if FETCH_INDEX exists
+            test_general_query(es, '*', 1)
```

Comment:
seems strange that you are using the same function for 3 different checks, please update this to separate functions